### PR TITLE
Whitelist restriction smartcontract deployment

### PR DESCRIPTION
**Pull request by 0xAleksaOpacic**
# Description
!IMPORTANT! This is just working Proof of Concept. Not official PR

Create a whitelist for contract deployment.
if the list is empty everyone can deploy a smart contract
if any address is in the list, only these acc can deploy

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual tests

1. Started network with 4 nodes

2. Deployed contract from one ACC (successful)

3. Added some random acc to white list:
go run main.go ibft contract_whitelist --addAccount 0x618b3B9D713276d793915D7418a527B9D8285109
Now you can only deploy from 0x618b3B9D713276d793915D7418a527B9D8285109 ACC

4. Restarted nodes

5. Tried to deploy from any other ACC (it will fail)

6. Added my acc to whitelist 

7. Restarted nodes

8. Now we can deploy onces again

# Additional comments

Fixes EDGE-24

### CODE REVIEW

**Comment by Kourin1996** [comment]
@Aleksao998 Just comment. It seems to be restrictions in TxPool, why you add command as subcommand of `ibft`?

**Comment by 0xAleksaOpacic** [comment]
> @Aleksao998 Just comment. It seems to be restrictions in TxPool, why you add command as subcommand of `ibft`?

When i started doing POC, the idea was to restrict it in the IBFT layer. But when I finished POC I realized its much better to restrict it on txpool layer. But because this is just POC, it was not worth changing command. When we decide to add this task i the sprint i will probably create new PR there are a lot of this that we can improve in this POC

File: txpool/txpool.go [hunk 1]
```diff
@@ -541,7 +545,25 @@ func (p *TxPool) validateTx(tx *types.Transaction) error {
...
+	fmt.Println("USAOOO")
+	fmt.Println("USAOOO")
+	fmt.Println("USAOOO")
+	fmt.Println("USAOOO")
+	fmt.Println("USAOOO")
+	fmt.Println("USAOOO")
+	fmt.Println("USAOOO")
+	fmt.Println("TX to", tx.To)
+
+	if !tx.CanApplyNewContract(p.features.ContractDeployment.WhiteList) {
```
Comment (lazartravica) [inline txpool/txpool.go:557]: You can move this check above, just after checking the tx signature

**Review by lazartravica — COMMENTED** [review]
LGTM, did you check that all paths for transaction addition go through `validateTx`?

**Comment by 0xAleksaOpacic** [comment]
> LGTM, did you check that all paths for transaction addition go through `validateTx`?

Yes, i am positive. Also double-checked with @dbrajovic who designed the architecture.

**Pull request closed without merging**